numpyro/distributions/continuous.py

```diff
@@ -3279,3 +3279,41 @@ def icdf(self, q: ArrayLike) -> ArrayLike:
     def sample(self, key, sample_shape: tuple[int, ...] = ()) -> jnp.ndarray:
         assert is_prng_key(key)
         return self.icdf(random.uniform(key, shape=self.shape(sample_shape)))
+
+    @property
+    def mean(self) -> ArrayLike:
+        safe_a = jnp.where(self.a > 1.0, self.a, 2.0)
+        return jnp.where(
+            self.a > 1.0,
+            self.b
+            * jnp.exp(
+                gammaln(1.0 - 1.0 / safe_a)
+                + gammaln(self.p + 1.0 / safe_a)
+                - gammaln(self.p)
```

Comment:
how about using https://docs.jax.dev/en/latest/_autosummary/jax.scipy.special.betaln.html for both mean and var?